### PR TITLE
Add regulars to potential train station names

### PR DESCRIPTION
**Pull request by mheguy**
Also.. we now increment the counter in the non-sorted `get_random` path.

### CODE REVIEW

**Comment by mheguy** [comment]
I really enjoy when people are surprised to see a name they recognize on stations, so let's make the odds a bunch lower. :)

File: features/train_station_names.lua [hunk 1]
```diff
@@ -1,17 +1,24 @@
 local Event = require 'utils.event'
 
+local function pick_name(event)
+    -- Create a weight table comprised of the backer name, a player's name, and a regular's name
+    local random_player = table.get_random(game.players, true)
```
Comment (grilledham) [inline features/train_station_names.lua:5]: Really minor point, but technically game.players isn't guaranteed to be a 'sorted'* table. There could be gaps in the indexes if we every use `game.remove_offline_players` which would mean `random_player` could be `nil`.

As that is extremely rare, I'd suggest just returning if `random_player` is nil

*by sorted I mean what `table.get_random` considers to be sorted.

File: utils/table.lua [hunk 2]
```diff
@@ -73,28 +73,34 @@ table.set = function(t, index, element)
...
 --- Chooses a random entry from a table
---@param t table to select an element from
---@param sorted boolean to indicate whether the table is sorted by numerical index or not
---@return a random element of table t
-table.get_random = function(t, sorted)
+-- @param t table to select an element from
+-- @param sorted boolean to indicate whether the table is sorted by numerical index or not
+-- @param key boolean to indicate whether to return the key or value
+-- @return a random element of table t
+table.get_random = function(t, sorted, key)
```
Comment (grilledham) [inline utils/table.lua:81]: I know this PR doesn't change this. But using `math.random` inside this function means the function can't be used outside of events. That should probably be mentioned in the comments.

File: utils/table.lua [hunk 3]
```diff
@@ -73,28 +73,34 @@ table.set = function(t, index, element)
...
--- @param item_index the index of the items, defaults to 1
--- @param weight_index the index of the weights, defaults to 2
--- @returns a random item with weighting
--- @see features.chat_triggers.hodor
+-- @param weight_table table of tables with items and their weights
+-- @param item_index number of the index of items, defaults to 1
+-- @param weight_index number of the index of the weights, defaults to 2
+-- @returns a table entry
+-- @see features.chat_triggers::hodor
 table.get_random_weighted = function(weighted_table, item_index, weight_index)
```
Comment (grilledham) [inline utils/table.lua:106]: same as above

File: features/train_station_names.lua (hunk 1 above)
Comment (mheguy) [reply features/train_station_names.lua:5]: Makes sense to me. And yes 'sorted' could use a better name.

File: utils/table.lua (hunk 2 above)
Comment (mheguy) [reply utils/table.lua:81]: Added

File: utils/table.lua (hunk 3 above)
Comment (mheguy) [reply utils/table.lua:106]: same as above

**Review by grilledham — APPROVED** [review]

**Pull request merged**